recipes/azure-identity-cpp/meta.yaml

```
@@ -0,0 +1,58 @@
+{% set name = "azure-identity-cpp" %}
+{% set version = "1.5.1" %}
+{% set sha256 = "651a660592540c267420b7e1c8dc98428a17396e45f7f6c92a864ab9a4fa028a" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/Azure/azure-sdk-for-cpp/archive/refs/tags/azure-identity_{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler("cxx") }}
+    - cmake
+    - make  # [not win]
+    - ninja  # [win]
+  host:
+    - azure-core-cpp >=1.9.0
+    - openssl
+  run:
+    - azure-core-cpp >=1.9.0
+    - openssl
+
+test:
+  commands:
+    - test -f "${PREFIX}/include/azure/identity.hpp"  # [not win]
+    - test -f "${PREFIX}/lib/libazure-identity${SHLIB_EXT}"  # [not win]
+    - test -f "${PREFIX}/share/azure-identity-cpp/azure-identity-cppTargets.cmake"  # [not win]
+    - if not exist %PREFIX%\\Library\\bin\\azure-identity.dll exit 1  # [win]
+    - if not exist %PREFIX%\\Library\\include\\azure\\identity.hpp exit 1  # [win]
+    - if not exist %PREFIX%\\Library\\lib\\azure-identity.lib exit 1  # [win]
+    - if not exist %PREFIX%\\Library\\share\\azure-identity-cpp\\azure-identity-cppTargets.cmake exit 1  # [win]
```

Please structure this a bit better (same for other recipes). Also, double back-slashes are not necessary.

```suggestion
    # shared libs
    - test -f "${PREFIX}/lib/libazure-identity.so"          # [linux]
    - test -f "${PREFIX}/lib/libazure-identity.dylib"       # [osx]
    - if not exist %LIBRARY_BIN%\azure-identity.dll exit 1  # [win]
    - if not exist %LIBRARY_BIN%\azure-identity.lib exit 1  # [win]

    # headers
    - test -f "${PREFIX}/include/azure/identity.hpp"  # [unix]
    - if not exist %LIBRARY_INC%\identity.hpp exit 1  # [win]

    # CMake metadata
    - test -f "${PREFIX}/share/azure-identity-cpp/azure-identity-cppTargets.cmake"                   # [unix]
    - if not exist %LIBRARY_PREFIX%\share\azure-identity-cpp\azure-identity-cppTargets.cmake exit 1  # [win]
```

---

It's also very weird to me that the CMake metadata is not in `$PREFIX/lib/cmake/<project>`, which is ~the standard~ much more common as a location.

---

Is this a problem? Are you suggesting we override it for the conda installation?